python/lsst/ap/association/loadDiaCatalogs.py

```diff
@@ -22,56 +22,97 @@
 """Task for pre-loading DiaSources and DiaObjects within ap_pipe.
 """
 import pandas as pd
-from sqlalchemy.exc import OperationalError, ProgrammingError
 
-import lsst.geom as geom
+import lsst.dax.apdb as daxApdb
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
-import lsst.sphgeom as sphgeom
+import lsst.pipe.base.connectionTypes as connTypes
+
 from lsst.utils.timer import timeMethod
 
-from lsst.ap.association.utils import convertTableToSdmSchema, readSchemaFromApdb
+from lsst.ap.association.utils import convertTableToSdmSchema, readSchemaFromApdb, getMidpointFromTimespan
 
 __all__ = ("LoadDiaCatalogsTask", "LoadDiaCatalogsConfig")
 
 
-class LoadDiaCatalogsConfig(pexConfig.Config):
+class LoadDiaCatalogsConnections(pipeBase.PipelineTaskConnections,
+                                 dimensions=("instrument", "group", "detector")):
+    regionTime = connTypes.Input(
+        doc="The predicted exposure region and time",
+        name="regionTimeInfo",
+        storageClass="RegionTimeInfo",
+        dimensions=("instrument", "group", "detector"),
+    )
+    diaObjects = connTypes.Output(
+        doc="DiaObjects preloaded from the APDB.",
+        name="preloaded_diaObjects",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+    diaSources = connTypes.Output(
+        doc="DiaSources preloaded from the APDB.",
+        name="preloaded_diaSources",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+    diaForcedSources = connTypes.Output(
+        doc="DiaForcedSources preloaded from the APDB.",
+        name="preloaded_diaForcedSources",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+
+
+class LoadDiaCatalogsConfig(pipeBase.PipelineTaskConfig,
+                            pipelineConnections=LoadDiaCatalogsConnections):
     """Config class for LoadDiaCatalogsConfig.
     """
+    apdb_config_url = pexConfig.Field(
+        dtype=str,
+        default=None,
+        optional=False,
+        doc="A config file specifying the APDB and its connection parameters, "
+            "typically written by the apdb-cli command-line utility. "
+            "The database must already be initialized.",
+    )
+
     pixelMargin = pexConfig.RangeField(
         doc="Padding to add to 4 all edges of the bounding box (pixels)",
         dtype=int,
         default=250,
         min=0,
     )
```

Comment:
Just noticed that this is dead code with the new algorithm. Do we still want to pad the incoming region?

Reply:
That's a very good point. I think we do want to have the ability to pad the region, so I'll modify this to a margin in arcseconds (since we won't have access to the pixel scale at this point), and use the dilated bounding circle of the region for the actual database query. Should I deprecate `pixelMargin`?

Reply:
I don't like the idea of slowing down our queries even further with a bounding circle 😬, but we can always optimize it later. 👍 for deprecation, just in case somebody is setting it.